packages/rest-client/src/main/RestClient.test.ts

```diff
@@ -130,6 +131,41 @@ describe('RestClient', () => {
       })
     })
 
+    it('can propagate api response status when desired', async () => {
+      nock('http://localhost:8080', {
+        reqheaders: { authorization: 'Bearer some_system_jwt' },
+      })
+        [method]('/api/test')
+        .reply(404, { message: 'some not found message' })
+
+      const call = restClient[method](
+        {
+          path: '/test',
+          headers: { header1: 'headerValue1' },
+          raw: true,
+          errorHandler: <ERROR>(path: string, verb: string, error: SanitisedError<ERROR>) => {
+            if (error.responseStatus === 404) {
+              const notFound = NotFound(`Resource not found on '${verb.toLowerCase()}: ${path}'`)
+              notFound.cause = error
```

Comment:
This isn't supported as part of the `http-errors` interface but 4 years ago it was slated for inclusion in version 3.. will wait to see if that ever happens. 
 
I'd like to see how bunyan handles `cause`, I'm guessing it probably doesn't but we could tweak the error handler to include details from the cause in the log output..